### PR TITLE
Add test for max coverage

### PR DESCRIPTION
**Pull request by cderv**
Adding remaining tests using mock functions when necessary and ignoring with 
`# nocov start `, `# nocov end`  when necessary

---
This change is [Reviewable](https://reviewable.io/reviews/rlesur/crrri/69)

### CODE REVIEW

**Comment by cderv** [comment]
We are at 91%. Not much. 

Some things went wrong with env var. I did not want to use `withr` to not have another suggests, but it would maybe make things smoother. 

Also, some part of the code are about chrome being reachable. Not so easy to test... 

I believe with 91% we are pretty good. 

We could decide do ignore (`# nocov`) some other part if we want better coverage badge because I think we won't be able to cover everything. 

This PR closes #45

File: R/http_methods.R [hunk 1]
```diff
@@ -118,7 +118,14 @@ inspect_target <- function(
 fetch_json <- function(host, port, secure, method, query = NULL) {
   check_host_port_args(host, port)
   url <- build_http_url(host, port, secure, path = c("json", method), query)
-  from_json(url)
+  tryCatch(
+    from_json(url),
+    error = function(e) {
+      rlang::abort(
+        message = sprintf("json protocol can't be reached at %s", url),
```
Comment (RLesur) [inline R/http_methods.R]:
```suggestion
        message = sprintf("json protocol cannot be reached at %s.", url),
```

File: R/utils.R [hunk 2]
```diff
@@ -99,7 +99,7 @@ parse_ws_url <- function(ws_url) {
   # NOTE: ws_url must be a character scalar
   ws_url <- httr::parse_url(ws_url)
   # ws_url scheme must be ws or wss:
-  if(!identical(ws_url$scheme, "ws") && !identical(ws_url, "wss")) {
+  if(!identical(ws_url$scheme, "ws") && !identical(ws_url$scheme, "wss")) {
```
Comment (RLesur) [inline R/utils.R]: Good catch!

File: tests/testthat/test-cdpsession.R [hunk 3]
```diff
@@ -32,6 +31,13 @@ test_that("port must be a numeric or a character scalar", {
 
 setup_chrome_test()
 
+test_that("ws url can be retrieve correctly", {
```
Comment (RLesur) [inline tests/testthat/test-cdpsession.R]:
```suggestion
test_that("ws url can be retrieved correctly", {
```

File: tests/testthat/test-chrome.R [hunk 4]
```diff
@@ -1,5 +1,34 @@
 context("test-chrome")
 
+test_that("we get the proxy env var correctly", {
+  skip_on_travis()
```
Comment (RLesur) [inline tests/testthat/test-chrome.R]: what is the problem if we do not skip this test on travis?

File: tests/testthat/test-chrome.R [hunk 5]
```diff
@@ -1,5 +1,34 @@
...
+  Sys.setenv(https_proxy = "dummy")
+  expect_identical(get_proxy(), c(https_proxy = "dummy"))
+  Sys.setenv(https_proxy = old)
+  old <- Sys.getenv("HTTP_PROXY")
+  Sys.setenv(HTTP_PROXY = "dummy")
+  expect_identical(get_proxy(), c(http_proxy = "dummy"))
+  Sys.setenv(HTTP_PROXY = old)
+})
+
+test_that("Proxy is correctly pass to chrome", {
```
Comment (RLesur) [inline tests/testthat/test-chrome.R]:
```suggestion
test_that("Proxy is correctly passed to chrome", {
```

File: tests/testthat/test-http_methods.R [hunk 6]
```diff
@@ -61,9 +71,39 @@ test_that("inspect_target() returns NULL for a correct target_id", {
...
 })
 
+test_that("is_localhost", {
+  expect_true(is_localhost("http://localhost:9222"))
+  expect_true(is_localhost("http://127.0.0.1:9222"))
+  expect_false(is_localhost("https://github.com/RLesur/crrri"))
+  expect_false(is_localhost(""))
+})
+
+test_that("browser_url use default function if no option set", {
```
Comment (RLesur) [inline tests/testthat/test-http_methods.R]:
```suggestion
test_that("browse_url use default function if no option set", {
```

File: tests/testthat/test-http_methods.R [hunk 7]
```diff
@@ -61,9 +71,39 @@ test_that("inspect_target() returns NULL for a correct target_id", {
...
+  expect_identical(
+    with_mock(
+      browseURL = identity,
+      expect_identical(browse_url(url), url)
+    ),
+    url)
+  options(viewer = old_opt)
+})
+
+test_that("browser_url use default function if no localhost", {
```
Comment (RLesur) [inline tests/testthat/test-http_methods.R]:
```suggestion
test_that("browse_url use default function if no localhost", {
```

File: R/utils.R (hunk 2 above)
Comment (cderv) [reply R/utils.R]: it helps finding it by creating tests ! ☺️

**Pull request merged**